processor/src/coin/bitcoin.rs

```diff
@@ -0,0 +1,376 @@
+            txid: one_transaction.txid,
+            vout: output_tx.n,
+            amount: output_tx.value.to_sat(),
+          }));
+        }
+      }
+    }
+    return Ok(outputs);
+  }
+
+  async fn prepare_send(
+    &self,
+    keys: ThresholdKeys<Secp256k1>,
+    transcript: RecommendedTranscript,
+    block_number: usize,
+    mut inputs: Vec<Output>,
+    payments: &[(Address, u64)],
+    change: Option<ProjectivePoint>,
+    fee: Fee,
+  ) -> Result<Self::SignableTransaction, CoinError> {
+    let mut vin_alt_list = Vec::new();
+    let mut vout_alt_list = Vec::new();
+    let change_addr = self.address(change.unwrap());
+
+    let mut input_sat = 0;
+    for one_input in &inputs {
+      input_sat += one_input.amount();
+      vin_alt_list.push(bitcoin::blockdata::transaction::TxIn {
+        previous_output: bitcoin::OutPoint { txid: one_input.0.txid, vout: one_input.0.vout },
+        script_sig: bitcoin::Script::new(),
+        sequence: bitcoin::Sequence(u32::MAX),
+        witness: bitcoin::Witness::default(),
+      });
+    }
+
+    let mut payment_sat = 0;
+    for one_payment in payments {
+      payment_sat += one_payment.1;
+      vout_alt_list.push(bitcoin::TxOut {
+        value: one_payment.1,
+        script_pubkey: one_payment.0.script_pubkey(),
+      });
+    }
+
+    let transaction_weight = MSignableTransaction::calculate_weight(vin_alt_list.len(), &payments[0].0, false);
+    let mut actual_fee = fee.calculate(transaction_weight);
+    let target_sat = actual_fee + payment_sat;
+    if input_sat < target_sat {
+      //TODO: Change it with TransactionError
+      return Err(CoinError::ConnectionError);
+    }
+    else if input_sat != target_sat {
+      let transaction_weight = MSignableTransaction::calculate_weight(vin_alt_list.len(), &payments[0].0, true);
+      actual_fee = fee.calculate(transaction_weight);
+      let target_sat = actual_fee + payment_sat;
+      if target_sat < input_sat {
+        let rest_sat = input_sat - target_sat;
+        vout_alt_list.push(bitcoin::TxOut {
+          value: rest_sat,
+          script_pubkey: change_addr.script_pubkey(),
+        });
+      }
+    }
+
+    let new_transaction = bitcoin::blockdata::transaction::Transaction {
+      version: 2,
+      lock_time: bitcoin::PackedLockTime(0),
+      input: vin_alt_list,
+      output: vout_alt_list,
+    };
+    let mut psbt = PartiallySignedTransaction::from_unsigned_tx(new_transaction.clone()).unwrap();
+    for (i, one_input) in (&inputs).iter().enumerate() {
+      let one_transaction = self.rpc.get_raw_transaction(&one_input.0.txid, None, None).await.unwrap();
+      let xonly_pubkey = XOnlyPublicKey::from_slice(&keys.group_key().to_encoded_point(true).x().to_owned().unwrap()).unwrap();
+      psbt.inputs[i].witness_utxo = Some(one_transaction.output[one_input.0.vout as usize].clone());
+      psbt.inputs[i].sighash_type = Some(PsbtSighashType::from_u32(SchnorrSighashType::All as u32));
```

Comment:
Explicit casts. For the latter, `SchnorrSighashType::All.into()` should work. Else, `PsbtSighashType::from(SchnorrSighashType::All)` definitely will.